### PR TITLE
[Merged by Bors] - refactor(Probability): Make kernels a type

### PR DESCRIPTION
**Pull request by YaelDillies**
Turn `kernel : AddSubmonoid (α → MeasureTheory.Measure β)` into `Kernel : Type _`. This is similar to #12386 but here to gain access to dot notation rather than for performance reasons.

---

A later PR will rename the predicates like `IsMarkovKernel`, `IsFiniteKernel`... to use dot notation.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Probability/Kernel/Basic.lean [hunk 1]
```diff
@@ -17,7 +17,7 @@ measurable sets `s` of `β`, `a ↦ κ a s` is measurable.
 ## Main definitions
 
 Classes of kernels:
-* `ProbabilityTheory.kernel α β`: kernels from `α` to `β`, defined as the `AddSubmonoid` of the
+* `ProbabilityTheory.Kernel α β`: kernels from `α` to `β`, defined as the `AddSubmonoid` of the
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/Basic.lean:20]: That docstring should be updated. It is not an AddSubmonoid anymore.

**Review by RemyDegenne — COMMENTED** [review]
The important changes are only in Kernel.Basic, right?

File: Mathlib/Probability/Kernel/Basic.lean [hunk 2]
```diff
@@ -329,10 +340,10 @@ end SFinite
 
 section Deterministic
 
-/-- Kernel which to `a` associates the dirac measure at `f a`. This is a Markov kernel. -/
-noncomputable def deterministic (f : α → β) (hf : Measurable f) : kernel α β where
-  val a := Measure.dirac (f a)
-  property := by
+/-- Kernel which to `a` associates the dirac measure at `f a`. This is a Markov Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/Basic.lean:343]:
```suggestion
/-- Kernel which to `a` associates the dirac measure at `f a`. This is a Markov kernel. -/
```

File: Mathlib/Probability/Kernel/Composition.lean [hunk 3]
```diff
@@ -588,51 +588,51 @@ variable {γ δ : Type*} {mγ : MeasurableSpace γ} {mδ : MeasurableSpace δ} {
 /-- The pushforward of a kernel along a measurable function.
 We include measurability in the assumptions instead of using junk values
 to make sure that typeclass inference can infer that the `map` of a Markov kernel
-is again a Markov kernel. -/
-noncomputable def map (κ : kernel α β) (f : β → γ) (hf : Measurable f) : kernel α γ where
-  val a := (κ a).map f
-  property := (Measure.measurable_map _ hf).comp (kernel.measurable κ)
+is again a Markov Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/Composition.lean:591]:
```suggestion
is again a Markov kernel. -/
```

File: Mathlib/Probability/Kernel/Composition.lean [hunk 4]
```diff
@@ -643,54 +643,54 @@ lemma map_const (μ : Measure α) {f : α → β} (hf : Measurable f) :
 /-- Pullback of a kernel, such that for each set s `comap κ g hg c s = κ (g c) s`.
 We include measurability in the assumptions instead of using junk values
 to make sure that typeclass inference can infer that the `comap` of a Markov kernel
-is again a Markov kernel. -/
-def comap (κ : kernel α β) (g : γ → α) (hg : Measurable g) : kernel γ β where
-  val a := κ (g a)
-  property := (kernel.measurable κ).comp hg
+is again a Markov Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/Composition.lean:646]:
```suggestion
is again a Markov kernel. -/
```

File: Mathlib/Probability/Kernel/CondDistrib.lean [hunk 5]
```diff
@@ -216,7 +216,7 @@ theorem condDistrib_ae_eq_condexp (hX : Measurable X) (hY : Measurable Y) (hs :
     exact @Measurable.ennreal_toReal _ (mβ.comap X) _ (measurable_condDistrib hs)
 
 /-- The conditional expectation of a function `f` of the product `(X, Y)` is almost everywhere equal
-to the integral of `y ↦ f(X, y)` against the `condDistrib` kernel. -/
+to the integral of `y ↦ f(X, y)` against the `condDistrib` Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/CondDistrib.lean:219]:
```suggestion
to the integral of `y ↦ f(X, y)` against the `condDistrib` kernel. -/
```

File: Mathlib/Probability/Kernel/CondDistrib.lean [hunk 6]
```diff
@@ -240,7 +240,7 @@ theorem condexp_prod_ae_eq_integral_condDistrib' [NormedSpace ℝ F] [CompleteSp
   · exact aestronglyMeasurable'_integral_condDistrib hX.aemeasurable hY hf_int.1
 
 /-- The conditional expectation of a function `f` of the product `(X, Y)` is almost everywhere equal
-to the integral of `y ↦ f(X, y)` against the `condDistrib` kernel. -/
+to the integral of `y ↦ f(X, y)` against the `condDistrib` Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/CondDistrib.lean:243]:
```suggestion
to the integral of `y ↦ f(X, y)` against the `condDistrib` kernel. -/
```

File: Mathlib/Probability/Kernel/CondDistrib.lean [hunk 7]
```diff
@@ -251,7 +251,7 @@ theorem condexp_prod_ae_eq_integral_condDistrib₀ [NormedSpace ℝ F] [Complete
   condexp_prod_ae_eq_integral_condDistrib' hX hY hf_int'
 
 /-- The conditional expectation of a function `f` of the product `(X, Y)` is almost everywhere equal
-to the integral of `y ↦ f(X, y)` against the `condDistrib` kernel. -/
+to the integral of `y ↦ f(X, y)` against the `condDistrib` Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/CondDistrib.lean:254]:
```suggestion
to the integral of `y ↦ f(X, y)` against the `condDistrib` kernel. -/
```

File: Mathlib/Probability/Kernel/Disintegration/Basic.lean [hunk 8]
```diff
@@ -169,27 +169,27 @@ Since every standard Borel space embeds measurably into `ℝ`, we can generalize
...
-A Borel space `Ω` embeds measurably into `ℝ` (with embedding `e`), hence we can get a `kernel α Ω`
-from a `kernel α ℝ` by taking the comap by `e`.
-Here we take the comap of a modification of `η : kernel α ℝ`, useful when `η a` is a probability
+/-- Auxiliary definition for `ProbabilityTheory.Kernel.condKernel`.
+A Borel space `Ω` embeds measurably into `ℝ` (with embedding `e`), hence we can get a `Kernel α Ω`
+from a `Kernel α ℝ` by taking the comap by `e`.
+Here we take the comap of a modification of `η : Kernel α ℝ`, useful when `η a` is a probability
 measure with all its mass on `range e` almost everywhere with respect to some measure and we want to
-ensure that the comap is a Markov kernel.
+ensure that the comap is a Markov Kernel.
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/Disintegration/Basic.lean:177]:
```suggestion
ensure that the comap is a Markov kernel.
```

File: Mathlib/Probability/Kernel/Disintegration/Basic.lean [hunk 9]
```diff
@@ -209,18 +209,18 @@ lemma borelMarkovFromReal_apply' (Ω : Type*) [Nonempty Ω] [MeasurableSpace Ω]
   · rw [Measure.comap_apply _ he.injective he.measurableSet_image' _ hs]
   · rw [Measure.comap_apply _ he.injective he.measurableSet_image' _ hs, Measure.dirac_apply]
 
-/-- When `η` is an s-finite kernel, `borelMarkovFromReal Ω η` is an s-finite kernel. -/
-instance instIsSFiniteKernelBorelMarkovFromReal (η : kernel α ℝ) [IsSFiniteKernel η] :
+/-- When `η` is an s-finite kernel, `borelMarkovFromReal Ω η` is an s-finite Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/Disintegration/Basic.lean:212]:
```suggestion
/-- When `η` is an s-finite kernel, `borelMarkovFromReal Ω η` is an s-finite kernel. -/
```

File: Mathlib/Probability/Kernel/Disintegration/Basic.lean [hunk 10]
```diff
@@ -209,18 +209,18 @@ lemma borelMarkovFromReal_apply' (Ω : Type*) [Nonempty Ω] [MeasurableSpace Ω]
...
+instance instIsSFiniteKernelBorelMarkovFromReal (η : Kernel α ℝ) [IsSFiniteKernel η] :
     IsSFiniteKernel (borelMarkovFromReal Ω η) :=
   IsSFiniteKernel.comapRight _ (measurableEmbedding_embeddingReal Ω)
 
-/-- When `η` is a finite kernel, `borelMarkovFromReal Ω η` is a finite kernel. -/
-instance instIsFiniteKernelBorelMarkovFromReal (η : kernel α ℝ) [IsFiniteKernel η] :
+/-- When `η` is a finite kernel, `borelMarkovFromReal Ω η` is a finite Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/Disintegration/Basic.lean:217]:
```suggestion
/-- When `η` is a finite kernel, `borelMarkovFromReal Ω η` is a finite kernel. -/
```

File: Mathlib/Probability/Kernel/Disintegration/Basic.lean [hunk 11]
```diff
@@ -209,18 +209,18 @@ lemma borelMarkovFromReal_apply' (Ω : Type*) [Nonempty Ω] [MeasurableSpace Ω]
...
+instance instIsFiniteKernelBorelMarkovFromReal (η : Kernel α ℝ) [IsFiniteKernel η] :
     IsFiniteKernel (borelMarkovFromReal Ω η) :=
   IsFiniteKernel.comapRight _ (measurableEmbedding_embeddingReal Ω)
 
-/-- When `η` is a Markov kernel, `borelMarkovFromReal Ω η` is a Markov kernel. -/
-instance instIsMarkovKernelBorelMarkovFromReal (η : kernel α ℝ) [IsMarkovKernel η] :
+/-- When `η` is a Markov kernel, `borelMarkovFromReal Ω η` is a Markov Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/Disintegration/Basic.lean:222]:
```suggestion
/-- When `η` is a Markov kernel, `borelMarkovFromReal Ω η` is a Markov kernel. -/
```

File: Mathlib/Probability/Kernel/Disintegration/Basic.lean [hunk 12]
```diff
@@ -470,28 +470,28 @@ section CountableOrCountablyGenerated
...
 irreducible_def condKernel [h : CountableOrCountablyGenerated α β]
-    (κ : kernel α (β × Ω)) [IsFiniteKernel κ] :
-    kernel (α × β) Ω :=
+    (κ : Kernel α (β × Ω)) [IsFiniteKernel κ] :
+    Kernel (α × β) Ω :=
   if hα : Countable α then condKernelCountable κ
   else letI := h.countableOrCountablyGenerated.resolve_left hα; condKernelBorel κ
 
-/-- `condKernel κ` is a Markov kernel. -/
+/-- `condKernel κ` is a Markov Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/Disintegration/Basic.lean:484]:
```suggestion
/-- `condKernel κ` is a Markov kernel. -/
```

File: Mathlib/Probability/Kernel/MeasureCompProd.lean [hunk 13]
```diff
@@ -30,22 +30,22 @@ open ProbabilityTheory
 namespace MeasureTheory.Measure
 
 variable {α β : Type*} {mα : MeasurableSpace α} {mβ : MeasurableSpace β}
-  {μ : Measure α} {κ η : kernel α β}
+  {μ : Measure α} {κ η : Kernel α β}
 
-/-- The composition-product of a measure and a kernel. -/
+/-- The composition-product of a measure and a Kernel. -/
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/MeasureCompProd.lean:35]:
```suggestion
/-- The composition-product of a measure and a kernel. -/
```

File: Mathlib/Probability/Kernel/WithDensity.lean [hunk 14]
```diff
@@ -9,21 +9,21 @@ import Mathlib.MeasureTheory.Integral.SetIntegral
 /-!
 # With Density
 
-For an s-finite kernel `κ : kernel α β` and a function `f : α → β → ℝ≥0∞` which is finite
+For an s-finite kernel `κ : Kernel α β` and a function `f : α → β → ℝ≥0∞` which is finite
 everywhere, we define `withDensity κ f` as the kernel `a ↦ (κ a).withDensity (f a)`. This is
-an s-finite kernel.
+an s-finite Kernel.
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/WithDensity.lean:14]:
```suggestion
an s-finite kernel.
```

File: Mathlib/Probability/Kernel/WithDensity.lean [hunk 15]
```diff
@@ -9,21 +9,21 @@ import Mathlib.MeasureTheory.Integral.SetIntegral
...
 
 ## Main definitions
 
-* `ProbabilityTheory.kernel.withDensity κ (f : α → β → ℝ≥0∞)`:
+* `ProbabilityTheory.Kernel.withDensity κ (f : α → β → ℝ≥0∞)`:
   kernel `a ↦ (κ a).withDensity (f a)`. It is defined if `κ` is s-finite. If `f` is finite
-  everywhere, then this is also an s-finite kernel. The class of s-finite kernels is the smallest
+  everywhere, then this is also an s-finite Kernel. The class of s-finite kernels is the smallest
```
Comment (RemyDegenne) [inline Mathlib/Probability/Kernel/WithDensity.lean:20]:
```suggestion
  everywhere, then this is also an s-finite kernel. The class of s-finite kernels is the smallest
```

**Review by RemyDegenne — APPROVED** [review]
Looks good!
bors d+

**Comment by YaelDillies** [comment]
Thanks for the quick review!

bors merge

**Pull request closed without merging**